### PR TITLE
[service.subtitles.subsceneplus] 1.0.64

### PR DESCRIPTION
**Pull request by Cih2001**
### Description
Subscene plus is a Kodi addon intended to facilitate downloading subtitles from subscene.com. All you need to do is to install it, set preferred languages to download subtitles, play a movie, go to subtitle menu, and download the subtitle.

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by Cih2001** [comment]
It is also worth to note that the current subscene addon on the official repository, is dated and not working anymore. That is the main reason to develop subscene plus.

link to the outdated plugin: https://kodi.wiki/view/Add-on:Subscene.com

File: service.subtitles.subsceneplus/addon.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  <requires>
+    <import addon="xbmc.python" version="2.26.0"/>
+    <import addon="script.module.requests" version="2.22.0"/>
+    <import addon="script.module.html5lib" version="0.999.0"/>
+    <import addon="vfs.libarchive" version="1.0.0"/>
+  </requires>
+  <extension point="xbmc.subtitle.module"
+             library="service.py" />
+  <extension point="xbmc.addon.metadata">
+    <summary lang="en">Subscene.com Subtitles</summary>
```
Comment (enen92) [inline service.subtitles.subsceneplus/addon.xml:15]: en_GB for summary and description

File: service.subtitles.subsceneplus/addon.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    <description lang="en"> A convenient addon to download subtitles from subscene.com. If you find it useful, please consider giving it a star in github. This is being actively maintained. In case of any problem, feel free to open an issue in github.
+    </description>
+    <platform>all</platform>
+    <license>GPL-2.0</license>
```
Comment (enen92) [inline service.subtitles.subsceneplus/addon.xml:19]: correct spdx identifiers are GPL-2.0-only or GPL-2.0-or-later

File: service.subtitles.subsceneplus/resources/lib/Subscene.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+if __name__ == "__main__":
+    # item = {
+    #     'title' : 'Yojimbo',
+    #     'year' : '1961' 
+    # }
+    # subtitles = SearchMovie(item)
+    # for subtitle in subtitles:
+    #     print (subtitle)
+    # DownloadSubtitle('/subtitles/joker-2019/english/2109631')
+    DownloadSubtitle('/subtitles/curse-of-the-golden-flower-man-cheng-jin-dai-huang-jin-jia/farsi_persian/446391')
```
Comment (enen92) [inline service.subtitles.subsceneplus/resources/lib/Subscene.py:263]: leftover?

File: service.subtitles.subsceneplus/service.py [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+SCRIPT_ID = ADD_ON.getAddonInfo('id')
+SCRIPT_NAME = ADD_ON.getAddonInfo('name').encode('utf-8')
+VERSION = ADD_ON.getAddonInfo('version')
+
+CWD = xbmc.translatePath(ADD_ON.getAddonInfo('path')).encode('utf-8')
+PROFILE = xbmc.translatePath(ADD_ON.getAddonInfo('profile')).encode('utf-8')
+RESOURCE = xbmc.translatePath(os.path.join(CWD, 'resources', 'lib')).encode('utf-8')
+TEMP = xbmc.translatePath(os.path.join(PROFILE, 'temp', '')).encode('utf-8')
+
+sys.path.append(RESOURCE)
```
Comment (enen92) [inline service.subtitles.subsceneplus/service.py:27]: please avoid using sys.path.append and import your file instead:

`from resources.lib.foo import bar`

File: service.subtitles.subsceneplus/service.py [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+        uniq_subs.append(v)
+
+    # sort uniq_subs based on their score.
+    uniq_subs.sort()
+            
+    for subtitle in uniq_subs:
+        listitem = xbmcgui.ListItem(
+            label = subtitle.lang,              # language name for the found subtitle
+            label2 = subtitle.name,             # file name for the found subtitle
+            iconImage = subtitle.rate(),        # rating for the subtitle, string 0-5
```
Comment (enen92) [inline service.subtitles.subsceneplus/service.py:170]: please do not use iconImage and thumbnailImage, they are both deprecated and were already removed from matrix. Use listitem.setArt({'icon': xxx, 'thumb': xxx, 'fanart': zzz, etc}

File: service.subtitles.subsceneplus/service.py [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+        urlpath = urllib.quote_plus(tmp_file)
+        (dirs, files) = xbmcvfs.listdir('archive://%s' % (urlpath))
+        # import web_pdb; web_pdb.set_trace()
+        for f in files:
+            src = 'archive://' + urlpath + '/' + f
+            dest = os.path.join(TEMP, f)
+            xbmcvfs.copy(src, dest)
+
+    else:
+        xbmc.executebuiltin(('XBMC.Extract("%s","%s")' % (tmp_file, TEMP)).encode('utf-8'), True)
```
Comment (enen92) [inline service.subtitles.subsceneplus/service.py:224]: drop XBMC. prefix. Also deprecated and already removed from matrix

**Comment by enen92** [comment]
Thanks for adjusting

**Comment by Cih2001** [comment]
Hi, Thanks for merging, Can you please let me know what's next? Is the official Kodi add-on repository updated or will it take some time?

**Pull request merged**